### ISSUE TITLE
heatmap generation

### ISSUE THREAD

**Issue by thanisornsr** [microsoft/human-pose-estimation.pytorch#163]
Hi, 

I am confusing about heatmap generation. 

1. From COCO dataset, the valid of joint are 0 for not in the image, 1 for in the image but not visible, and 2 for in the image and visible. How do you encode this into heatmap? If valid is 0, it will be zero matrix? if it's 1 or 2, it will be heatmap with 1 peak? Am I right?

2. The loss function when training, It's MSE between predicted heatmaps and targeted heatmaps according to the paper. Do I have to multiply with valid? or it's not needed?

All the best,
Than

**Comment by tucachmo2202** [comment]
> * From COCO dataset, the valid of joint are 0 for not in the image, 1 for in the image but not visible, and 2 for in the image and visible. How do you encode this into heatmap? If valid is 0, it will be zero matrix? if it's 1 or 2, it will be heatmap with 1 peak? Am I right?
> *

Hi, They use 2D Gaussian centered to generate heatmap.
